src/shogun/util/factory.h

```diff
@@ -130,6 +131,41 @@ namespace shogun
 		return result;
 	}
 
+	template <class T>
+	std::shared_ptr<Features> sparse_features(SGMatrix<T> mat)
+	{
+		return std::make_shared<SparseFeatures<T>>(mat);
+	}
+
+	std::shared_ptr<Features> sparse_features(std::shared_ptr<File> file, EPrimitiveType primitive_type = PT_FLOAT64)
+	{
+		require(file, "No file provided.");
+		std::shared_ptr<Features> result = nullptr;
```

Comment:
again as said above this should be within features factory function.
for that to be able to work you'll have to do a bit of dispatching, i.e. to detect the type of File object being passed (csv or libsvmfile) and based on that create the right feature type object (dense or sparse). i'd use type_index(typeid()) to detect the type of file that has been passed over, but you can even use get_name() and do a string comparison.